### DISCUSSION TITLE
Generic collection fixtures

### DISCUSSION THREAD

**Discussion by lonix1** [xunit/xunit#2425] [Question] [answered]
@bradwilson I didn't know whether to file this as an issue or discussion, sorry!

A generic fixture and support classes:

````cs
public class Fixture<T> : IAsyncLifetime, IDisposable {
  public Fixture() { }
  public void Dispose() { }
  public Task InitializeAsync() => Task.CompletedTask;
  public Task DisposeAsync() => Task.CompletedTask;
}


[CollectionDefinition("collection")]
public class FixtureCollection<T> : ICollectionFixture<Fixture<T>> { }


[Collection("collection")]
public abstract class TestBase<T> {
  protected TestBase(Fixture<T> fixture) => _fixture = fixture;
  protected readonly Fixture<T> _fixture;
}
````

I stripped down those classes to show a minimal sample (but there is "stuff" in them).

And the test class:
````cs
public class Foo { }


public class Tests : TestBase<Foo> {
  public Tests(Fixture<Foo> fixture) : base(fixture) { }
  [Fact] public void Test1() { }
  [Fact] public void Test2() { }
}
````

Tests fail:
> The following constructor parameters did not have matching fixture data: Fixture`1 fixture

If I make this change then it works, but I can't use it as a collection fixture:
````cs
//[Collection("collection")]                                      // removed
public abstract class TestBase<T> : IClassFixture<Fixture<T>> {   // added IClassFixture
````

I read [the docs](https://xunit.net/docs/shared-context).

**Is this use case supported?**

**Comment by bartelink** [comment]
ref [CC of this SO question](https://stackoverflow.com/questions/70090819/generic-collection-fixtures-in-xunit?noredirect=1#comment123911515_70090819) - the question in particular is related to the fact that a `CollectionAttribute` on a _generic_ base class of the TestClass is not picked up (although a basic confirmation that Collection Fixtures must be concrete is also helpful as a double confirm)

**Comment by bradwilson** [comment, marked as answer]
The answer is: it's not a supported use case, currently. Feel free to open an issue to get it added to v3.

**Comment by lonix1** [comment]
Thanks - I used the "Create issue from discussion" feature. Issue #2557.
